packages/common/src/hooks/useDownload.ts

```diff
@@ -0,0 +1,122 @@
+import { useEffect, useState } from 'react'
+
+import type { AudiusSdk } from '@audius/sdk'
+import { isEqual } from 'lodash'
+import { shallowEqual, useSelector } from 'react-redux'
+import { usePrevious } from 'react-use'
+
+import { DownloadQuality } from '~/models'
+import dayjs from '~/utils/dayjs'
+import { encodeHashId } from '~/utils/hashIds'
+
+import { ID } from '../models/Identifiers'
+import { Track, StemTrack } from '../models/Track'
+import { getTrack, getTracks } from '../store/cache/tracks/selectors'
+import { CommonState } from '../store/commonStore'
+import { getCurrentUploads } from '../store/stems-upload/selectors'
+
+const sortByDateAsc = (a: Track, b: Track) =>
+  dayjs(a.created_at).diff(dayjs(b.created_at))
+
+export const useCurrentStems = ({ trackId }: { trackId: ID }) => {
```

Comment:
nit: I know this is just a file rename with some stuff cut out, but would be nice to have jsdoc for each of these hooks